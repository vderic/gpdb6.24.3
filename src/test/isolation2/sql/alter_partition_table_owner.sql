-- Test that ALTER TABLE ... OWNER TO recurses to child partitions when run on the root partition in utility mode for coordinator.
CREATE ROLE testrole;
CREATE TABLE p_alterowner_1 (id INTEGER, name TEXT) DISTRIBUTED BY (id) PARTITION BY RANGE(id) (START(1) END(3) EVERY(1));

-1U: ALTER TABLE p_alterowner_1 OWNER TO testrole;
-1U: SELECT c.relname, pg_catalog.pg_get_userbyid(c.relowner) AS owner FROM pg_class c WHERE relname like 'p_alterowner_1%';

DROP TABLE p_alterowner_1;
DROP ROLE testrole;
