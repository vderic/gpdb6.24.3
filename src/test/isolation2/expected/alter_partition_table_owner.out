-- Test that ALTER TABLE ... OWNER TO recurses to child partitions when run on the root partition in utility mode for coordinator.
CREATE ROLE testrole;
CREATE
CREATE TABLE p_alterowner_1 (id INTEGER, name TEXT) DISTRIBUTED BY (id) PARTITION BY RANGE(id) (START(1) END(3) EVERY(1));
CREATE

-1U: ALTER TABLE p_alterowner_1 OWNER TO testrole;
ALTER
-1U: SELECT c.relname, pg_catalog.pg_get_userbyid(c.relowner) AS owner FROM pg_class c WHERE relname like 'p_alterowner_1%';
 relname                | owner    
------------------------+----------
 p_alterowner_1         | testrole 
 p_alterowner_1_1_prt_1 | testrole 
 p_alterowner_1_1_prt_2 | testrole 
(3 rows)

DROP TABLE p_alterowner_1;
DROP
DROP ROLE testrole;
DROP
